src/main/java/net/minecraftforge/common/advancements/CustomAdvancementRewards.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.advancements;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.util.GsonHelper;
+import net.minecraftforge.registries.ForgeRegistries;
+
+/**
+ * Class holding the hooks and registry for custom advancement rewards
+ */
+public final class CustomAdvancementRewards {
+
+    /**
+     * Hook for AdvancementRewards#grant
+     */
+    public static void grant(ServerPlayer player, ICustomAdvancementReward[] customRewards)
+    {
+        for (ICustomAdvancementReward reward : customRewards)
+        {
+            reward.grant(player);
+        }
+    }
+
+    /**
+     * Hook for AdvancementRewards#serialize
+     */
+    public static void serialize(JsonObject jsonObject, ICustomAdvancementReward[] customRewards)
+    {
+        if (customRewards.length == 0) return;
+        JsonArray arr = new JsonArray();
+        for (ICustomAdvancementReward customReward : customRewards)
+        {
+            @SuppressWarnings("unchecked")
+            ICustomAdvancementReward.Serializer<ICustomAdvancementReward> serializer = (ICustomAdvancementReward.Serializer<ICustomAdvancementReward>) customReward.getSerializer();
```

Comment:
Remove this cast by making the signature of getSerializer: `<T extends ICustomAdvancementReward> Serializer<T> getSerializer();`